solidity/contracts/liquidity-protection/LiquidityProtectionStats.sol

```diff
@@ -0,0 +1,131 @@
+     */
+    function increaseTotalAmounts(
+        address _provider,
+        IDSToken _poolToken,
+        IERC20Token _reserveToken,
+        uint256 _poolAmount,
+        uint256 _reserveAmount
+    ) external override ownerOnly {
+        totalPoolAmount[_poolToken] = totalPoolAmount[_poolToken].add(_poolAmount);
+        totalReserveAmount[_poolToken][_reserveToken] = totalReserveAmount[_poolToken][_reserveToken].add(_reserveAmount);
+        totalProviderAmount[_poolToken][_reserveToken][_provider] = totalProviderAmount[_poolToken][_reserveToken][_provider].add(_reserveAmount);
+    }
+
+    /**
+     * @dev decreases the total amounts
+     *
+     * @param _provider         liquidity provider address
+     * @param _poolToken        pool token address
+     * @param _reserveToken     reserve token address
+     * @param _poolAmount       pool token amount
+     * @param _reserveAmount    reserve token amount
+     */
+    function decreaseTotalAmounts(
+        address _provider,
+        IDSToken _poolToken,
+        IERC20Token _reserveToken,
+        uint256 _poolAmount,
+        uint256 _reserveAmount
+    ) external override ownerOnly {
+        totalPoolAmount[_poolToken] = totalPoolAmount[_poolToken].sub(_poolAmount);
+        totalReserveAmount[_poolToken][_reserveToken] = totalReserveAmount[_poolToken][_reserveToken].sub(_reserveAmount);
+        totalProviderAmount[_poolToken][_reserveToken][_provider] = totalProviderAmount[_poolToken][_reserveToken][_provider].sub(_reserveAmount);
+    }
+
+    /**
+     * @dev resets the total amounts
+     *
+     * @param _provider         liquidity provider address
+     * @param _poolToken        pool token address
+     * @param _reserveToken     reserve token address
+     */
+    function resetTotalAmounts(
+        address _provider,
+        IDSToken _poolToken,
+        IERC20Token _reserveToken
+    ) external override ownerOnly {
+        totalPoolAmount[_poolToken] = 0;
+        totalReserveAmount[_poolToken][_reserveToken] = 0;
+        totalProviderAmount[_poolToken][_reserveToken][_provider] = 0;
+    }
+
+    function seed(
+        address[] memory _tokens,
+        address[] memory _reserve0s,
+        address[] memory _reserve1s,
+        address[] memory _providers,
+        uint256[] memory _poolAmounts,
+        uint256[] memory _reserve0Amounts,
+        uint256[] memory _reserve1Amounts,
+        uint256[] memory _provider0Amounts,
+        uint256[] memory _provider1Amounts
+    ) external seederOnly {
+        uint256 length = _tokens.length;
+        require(length == _reserve0s.length);
+        require(length == _reserve1s.length);
+        require(length == _providers.length);
+        require(length == _poolAmounts.length);
+        require(length == _reserve0Amounts.length);
+        require(length == _reserve1Amounts.length);
+        require(length == _provider0Amounts.length);
+        require(length == _provider1Amounts.length);
```

Comment:
You can do all these checks in a single `require` (especially if there is no/same error message)

Reply:
Fixed.